core/src/smartcontracts/isi/mod.rs

```diff
@@ -40,6 +40,125 @@ pub trait Registrable {
     fn build(self, authority: &AccountId) -> Self::Target;
 }
 
+impl ValidInstruction for InstructionBox {
+    fn len(&self) -> usize {
+        use InstructionBox::*;
+
+        match self {
+            Register(register_box) => register_box.len(),
+            Unregister(unregister_box) => unregister_box.len(),
+            Mint(mint_box) => mint_box.len(),
+            Burn(burn_box) => burn_box.len(),
+            Transfer(transfer_box) => transfer_box.len(),
+            If(if_box) => if_box.len(),
+            Pair(pair_box) => pair_box.len(),
+            Sequence(sequence) => sequence.len(),
+            Fail(fail_box) => fail_box.len(),
+            SetKeyValue(set_key_value) => set_key_value.len(),
+            RemoveKeyValue(remove_key_value) => remove_key_value.len(),
+            Grant(grant_box) => grant_box.len(),
+            Revoke(revoke_box) => revoke_box.len(),
+            ExecuteTrigger(execute_trigger) => execute_trigger.len(),
+            SetParameter(set_parameter) => set_parameter.len(),
+            NewParameter(new_parameter) => new_parameter.len(),
+            Upgrade(upgrade_box) => upgrade_box.len(),
+        }
+    }
+}
+
+impl ValidInstruction for SetKeyValueBox {
```

Comment:
Probably better done in a macro.

Reply:
Also, do we really need this trait? Can't we have a single `fn` with a top-level match statement? 

Because we're not really using dynamic dispatch (even though if we did, LLVM devirtualiser would still emplace the function calls), and as mentioned above, we're not really checking that the correct `len` is implemented.